corehq/apps/app_manager/suite_xml.py

```diff
@@ -1615,7 +1615,8 @@ def entries(self):
                     )
                 )
                 if isinstance(module, Module):
-                    for datum_meta in self.get_datum_meta_module(module, use_filter=False):
+                    use_filter = False if module.has_no_forms() else bool(module.case_details.short.filter)
```

Comment:
Is this right. Isn't the logic we want the following:
* add filter only if the module has not forms and the module has case filtering configured

`use_filter = module.has_no_forms() and bool(module.case_details.short.filter)`